components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -947,17 +973,60 @@ private boolean isUsernameCaseSensitive(String tenantQualifiedUsername) {
      * @return
      * @throws IdentityOAuth2Exception
      */
+    /**
+     * @deprecated use {@link #getOIDCIdTokenAudiences(String, String )} instead.
+     */
+    @Deprecated
     public List<String> getOIDCAudiences(String tenantDomain, String consumerKey) throws IdentityOAuth2Exception {
 
-        List<String> audiences = new ArrayList<>();
+        return this.getOIDCIdTokenAudiences(tenantDomain, consumerKey);
+
+    }
+
+    /**
+     * Retrieves OIDC ID Token audience values configured for an oauth consumer app.
+     *
+     * @param tenantDomain application tenant domain
+     * @param consumerKey  client ID
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public List<String> getOIDCIdTokenAudiences(String tenantDomain, String consumerKey)
+            throws IdentityOAuth2Exception {
+
+        List<String> idTokenAudiences = new ArrayList<>();
+        idTokenAudiences = getAudiencesFromDB(idTokenAudiences, tenantDomain, consumerKey);
+        return idTokenAudiences;
```

Comment:
`List<String> idTokenAudiences = new ArrayList<>();` can be directly defined inside `getAudiencesFromDB` method as `List<String> audiences = new ArrayList<>();`
Then we can directly `return getAudiencesFromDB(tenantDomain, consumerKey);`